File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r, profit = 0, 1, 0

        while r < len(prices):
            profit = max(prices[r] - prices[l], profit)
            if prices[l] > prices[r]:
                l += 1
                r = l
            r += 1

        return profit

            

File: Data Structures & Algorithms/task-scheduling/submission-1.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        vector<int> freq(26);
        for (char task : tasks) {
            ++freq[task - 'A'];
        }
        
        int maxf = *max_element(freq.begin(), freq.end());
        int maxCount = 0;
        for (int i : freq) {
            if (i == maxf) {
                maxCount++;
            }
        }

        int time = (maxf - 1) * (n + 1) + maxCount;
        return max((int)tasks.size(), time);

    }
};
